databroker/src/glob.rs

```diff
@@ -19,6 +19,29 @@ pub enum Error {
     RegexError,
 }
 
+pub fn to_glob_string(glob: &str) -> String {
```

Comment:
I think it would be better to have not return a "glob string" from `crate::glob`. Instead, the implementation, i.e. the usage of `glob_match` or such implementation detail should be handled inside this file and only a `glob::Matcher` or something equivalent should be returned for use by external users.

PS. The same thing applies to `to_regex_string`

Reply:
Done, all is handled now in the glob file.

Reply:
It's still returning a "glob string" which is just a `String`.
And then using `glob_string_match(glob: &str, path: &str)` on that "glob string".

My point was that it would be better to have an API like this:
```rust
// in glob.rs

impl Matcher {
  pub fn new(glob: &str) -> Result<Matcher, Error> {

  }

  pub fn is_match(path: &str) {

  }
}
```

so that we don't pass things called "glob strings" around that are no different than `String`s.

That would make it easier to use (and harder to accidentally get the strings wrong) since it would be type safe.